### ISSUE TITLE
[BUG][CML]Sound setting shows Dummy and firmware load failed(intermitent)

### ISSUE THREAD

**Issue by YvonneYang2** [thesofproject/linux#998]
**Describe the bug**
Sound setting shows Dummy, "aplay -l" cannot show PCM device list.

**To Reproduce**

Do system reboot 

**Frequency**

once/twice a day

**Expected result**
Sound setting shows normally

**Actual result**
"aplay -l" cannot show PCM device list, sound setting shows "Dummy". 

**Environment**
Rander UCM https://github.com/RanderWang/intel-UCM/commits/hdmi
Firmware: 6cc8da10 https://github.com/thesofproject/sof/commits/cml-hda-dmic-001-drop-stable
Kernel: 0ce7065 https://github.com/thesofproject/linux/tree/release/sof-v5.0 
Topology: file: tools/topology/sof-hda-generic.tplg, sof-hda-generic-2ch.tplg same as firmware
platform ：CML laptop 

**All logs**
reboot with HDMI connected 
[Dummy_logger.log](https://github.com/thesofproject/linux/files/3240534/Dummy_logger.log)
[pulseverbose.log](https://github.com/thesofproject/linux/files/3240535/pulseverbose.log)
[Dummy_dmesg.log](https://github.com/thesofproject/linux/files/3240536/Dummy_dmesg.log)

**Comment by bardliao** [comment]
From the pulseverbose.log 
```
(   0.368|   0.002) I: [pulseaudio] module-udev-detect.c: Found 0 cards.
```
It indicates that pulse audio can't find the sound card. And the dmesg shows 
```
[    2.313063] HDMI HDA Codec ehdaudio0D2: HDMI: failed to get afg sub nodes
[    2.313104] HDMI HDA Codec ehdaudio0D2: Failed in parse and map nid with err: -22
[    2.313111] HDMI HDA Codec: probe of ehdaudio0D2 failed with error -22
```
I also can reproduce this issue with log [dmesg_codec_probe_error.txt](https://github.com/thesofproject/linux/files/3240448/dmesg_codec_probe_error.txt)
And it is failed at HDA codec.
```
[   11.780742] hdaudio ehdaudio0D0: cannot read sub nodes for FG 0x10
[   11.783091] sof-audio-pci 0000:00:1f.3: device init failed for hdac device
[   11.785496] sof-audio-pci 0000:00:1f.3: error: codec #0 probe error, ret: -22
```
The error is not exactly the same but all related to hda.
The issue seems can only be reproduced with release/sof-v5.0 at commit 0ce7065, but can't reproduce on the latest release/sof-v5.0.

So I think the issue may be fixed by #994 and/or #995 which both fix spin lock issue on hda.
I suggest that we can close the issue until we see it again.

**Comment by YvonneYang2** [comment]
Issue not reproduced with the test recipe below, so close the bug until we see it again.

Rander UCM https://github.com/RanderWang/intel-UCM/commits/hdmi
Firmware: 6cc8da10 https://github.com/thesofproject/sof/commits/cml-hda-dmic-001-drop-stable
Kernel: 0ce7065 https://github.com/thesofproject/linux/tree/release/sof-v5.0 
Topology: file: tools/topology/sof-hda-generic.tplg, sof-hda-generic-2ch.tplg same as firmware
platform ：CML laptop with codec ALC3204
